kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/KinClient.java

```diff
@@ -215,7 +230,6 @@ public void deleteAccount(int index) throws DeleteAccountException {
     /**
      * Deletes all accounts.
      */
-    @SuppressWarnings("WeakerAccess")
     public void clearAllAccounts() {
         keyStore.clearAllAccounts();
```

Comment:
don't we need a `loadAccount` here as well?

Reply:
I added loadAccount in a "lazy" way, meaning only when someone acctualy tries to do something, meaning getAccount, getAccountCount and getAccountByPublicAddress.
Do you think we must add it also to clearAllAccounts, deleteAccount and addKeyPair?

Reply:
I think it's a potential bug, if you delete an account on kinClient1, while kinClient2 added an account, the last will win, I would expect one account to be added and one be deleted.

Reply:
but at the second one of the kinClient will try to use one of the methods that query the account(getAccount or getAccountCount or getAccountByPublicAddress) then he will get the correct result and the internal list will be updated 
**for example:**
current account in database are("1", "2", "3", 4") then kinClient1 delete account "2" and then kinClient2 add account "5". the database will now look like this ("1", "3", "4", "5").
Although the internal list for each one of them will be different,  at the second that one of them will invoke one of the above methods then the internal list of both of them will be updated.

I added another test for this case.
What is your concern exactly? or you don't like the idea that the internal state is temporary not the same?

Reply:
@yosriz please review again after my last changes